openmetadata-service/src/main/java/org/openmetadata/service/resources/dqtests/TestCaseResultResource.java

```diff
@@ -0,0 +1,435 @@
+package org.openmetadata.service.resources.dqtests;
+
+import static org.openmetadata.service.Entity.TEST_CASE;
+
+import io.swagger.v3.oas.annotations.ExternalDocumentation;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.ExampleObject;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Set;
+import javax.json.JsonPatch;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.api.tests.CreateTestCaseResult;
+import org.openmetadata.schema.tests.TestCase;
+import org.openmetadata.schema.tests.type.TestCaseResult;
+import org.openmetadata.schema.tests.type.TestCaseStatus;
+import org.openmetadata.schema.type.Include;
+import org.openmetadata.schema.type.MetadataOperation;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.jdbi3.TestCaseResultRepository;
+import org.openmetadata.service.resources.Collection;
+import org.openmetadata.service.resources.EntityTimeSeriesResource;
+import org.openmetadata.service.resources.feeds.MessageParser;
+import org.openmetadata.service.search.SearchListFilter;
+import org.openmetadata.service.search.SearchSortFilter;
+import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.security.policyevaluator.OperationContext;
+import org.openmetadata.service.security.policyevaluator.ResourceContextInterface;
+import org.openmetadata.service.security.policyevaluator.TestCaseResourceContext;
+import org.openmetadata.service.util.EntityUtil;
+import org.openmetadata.service.util.RestUtil;
+import org.openmetadata.service.util.ResultList;
+
+@Slf4j
+@Path("/v1/dataQuality/testCases/testCaseResults")
```

Comment:
Since test case result is always mapped to a specific test case. It might make more sense to parametrize it at the [resource level.](https://www.dropwizard.io/en/stable/manual/core.html#resources)

```suggestion
@Path("/v1/dataQuality/testCases/{fqn}/testCaseResults")
```

Reply:
This is handled in the resource itself. 
```
@GET
   @Path("/{fqn}")
```

We have CRUD operations for a few thing, but we should allow user to list all test case results (without a test case itself)